src/builder.ts

```diff
@@ -1413,14 +1437,21 @@ export class SchemaBuilder {
     })
   }
 
-  addDynamicOutputMembers(block: OutputDefinitionBlock<any>, isList: boolean, stage: 'walk' | 'build') {
+  addDynamicOutputMembers(
+    block: OutputDefinitionBlock<any>,
+    wrapping: NexusWrapKind[],
+    stage: 'walk' | 'build'
+  ) {
     eachObj(this.dynamicOutputFields, (val, methodName) => {
       if (typeof val === 'string') {
         return this.addDynamicScalar(methodName, val, block)
       }
       // @ts-ignore
       block[methodName] = (...args: any[]) => {
-        const config = isList ? [args[0], { list: isList, ...args[1] }] : args
+        const wrappedType =
+          args[1]?.type !== undefined ? wrapAsNexusType(args[1].type, wrapping, false) : undefined
+        const config = wrappedType !== undefined ? [args[0], { ...args[1], type: wrappedType }] : args
+
```

Comment:
Same here, not so sure what to do now @tgriesser

Reply:
Makes me think this is actually wrong and need to be updated anyway, the `nonNullDefault` is always put to false, but my guts tell me this should be handled differently anyway